add pylint tests

bilelmoussaoui opened the pull request:
The structure of tests isn't that perfect at the moment. This PR is more to force Travis to detect the working env(.travis.yml)

bilelmoussaoui committed:
add pylint tests

bilelmoussaoui committed:
mark the pylint.sh as exec & add a nice badge

bilelmoussaoui committed:
remove typo space

bilelmoussaoui committed:
use env sh

Foggalong commented:
(@bil-elmoussaoui you're gonna have to let me know when this is working. This is my first venture into using Travis for anything so I'm gonna be picking this up as we go along)

bilelmoussaoui commented:
The pylint part is already working as expected :) but let's wait until i implement the json linting too (which should be done asap to avoid PR like this #3884)

Foggalong commented:
Been thinking about this while I was away: would you be opposed to using pep8 instead of pylint? It seems to filter out the slightly daft rules that we'd ignore anyway but still does a good job of keeping the style consistent.

Let me know too if there's anything you need me to do on the account end :)

bilelmoussaoui committed:
Merge branch 'master' of https://github.com/numixproject/numix-core into tests/pylint

bilelmoussaoui committed:
first attempt on json validation

bilelmoussaoui committed:
fix pylint issues

bilelmoussaoui committed:
fix json.JSONDecodeError

bilelmoussaoui committed:
add schema tests to travis

bilelmoussaoui committed:
install pip packages locally

bilelmoussaoui committed:
--user is not possible on virtualenv

bilelmoussaoui committed:
JSONDecodeError is only available from python3.5+

bilelmoussaoui committed:
use python3 from env

bilelmoussaoui committed:
fix schema, finally working

bilelmoussaoui committed:
replace pylint with pep8

bilelmoussaoui commented:
@Foggalong It's ready now :) 
See what's done here:
- Switched from pylint to pep8
- Fixed few too long lines (pep8 issues)
- Introduce Travis file (.travis.yml)
- Add schema.py & schema.json, the schema.json contains rules of how the data.json file should be created, it will throw errors if the root is missing, or if it contains a duplicated item on symlinks and a lot of other stuff. It also checks for Android. The schema.py file is the file testing that reads the database and validates it using the schema. It will throw a red error message if there are any schema or JSON syntax errors. You can test that manually by cloning this branch, and installing python3-jsonschema (available in a lot of distros) or just `sudo pip3 install jsonschema` and then run from the numix-core directory (`python3 ./tests/schema.py`)

This can now be merged. Other tests will be introduced in the future;
I'm going to add later, a check to be sure that the database is correctly sorted, and we will see later for validating the icons source (we should first decide which tool to use to optimize the icons, and do that for every icon out there)

bilelmoussaoui commented on tests/schema.json:
```diff
@@ -0,0 +1,45 @@
+            },
+            "additionalProperties": false,
+            "minProperties": 1,
+            "required": [
```
@Foggalong The required Linux here will be removed later once we have the Android icons in the data.json file; For now I kept the Linux here to be sure that everything is validated

bilelmoussaoui commented on README.md:
```diff
@@ -1,4 +1,6 @@
 # Numix Core
+[![Build Status](https://travis-ci.org/numixproject/numix-core.svg?branch=master)](https://travis-ci.org/numixproject/numix-core)
```
@Foggalong Don't worry about the unknown status here, as the badge shows the status of the master branch, and as the master branch doesn't contain any .travis.yml file it shows this badge; but once the merge is done, we will see a beautiful green one :)
